invenio_users_resources/ext.py

```diff
@@ -9,11 +9,36 @@
 
 """Invenio module providing management APIs for users and roles/groups."""
 
+from invenio_accounts.models import User
+from invenio_userprofiles.models import UserProfile
+from sqlalchemy import event
+
 from . import config
 from .resources import GroupsResource, GroupsResourceConfig, UsersResource, \
     UsersResourceConfig
 from .services import GroupsService, GroupsServiceConfig, UsersService, \
     UsersServiceConfig
+from .utils import reindex_user, unindex_user
+
+
+@event.listens_for(User, "after_insert")
+@event.listens_for(User, "after_update")
+def after_user_update(mapper, connection, target):
+    """Reindex the user after an insert or update to DB."""
+    reindex_user(target)
+
+
+@event.listens_for(UserProfile, "after_insert")
+@event.listens_for(UserProfile, "after_update")
+def after_userprofile_update(mapper, connection, target):
+    """Reindex the user after their profile has changed."""
+    reindex_user(target.user)
```

Comment:
it seemed to be overkill to go ahead and ship off a celery task for re-indexing this one updated user aggregate